discovery-infra/utils.py

```diff
@@ -16,6 +17,24 @@
 conn = libvirt.open("qemu:///system")
 
 
+def on_exception(*, message=None, callback=None, silent=False):
```

Comment:
the callback and silent arguments are not in use in this pr. 
multiple prs are sharing this decorator so I left it as is.